pennylane/ops/qubit/parametric_ops_multi_qubit.py

```diff
@@ -736,58 +885,314 @@ def compute_eigvals(*params: TensorLike, **hyperparams) -> TensorLike:
+        :math:`|j\rangle\langle j|` onto the :math:`d` consecutive smallest (largest, for
+        :math:`\sigma=-1`) basis states,
+        into projectors that contain a power of two basis state projectors. We will see below that
+        these are exactly the generators of (multi-controlled) phase shift gates, allowing us
+        to implement the decomposition. As we may flip the sign of these individual phase shift
+        gates, we may also use subtraction when decomposing :math:`\Pi`.
+
+        This decomposition of :math:`\Pi` is based on decomposing the integer :math:`d` into sums
+        and differences of powers of two, which is detailed in a separate section below. This
+        decomposition reads
+
+        .. math::
+
+            d = \sum_{i=0}^{n-1} c_i 2^{n-1-i}, \quad c_i\in\{-1, 0, 1\}.
+
+        Now we need to use this decomposition to combine projectors of :math:`2^{n-1-i}`
+        computational basis state projectors into the desired :math:`\Pi`.
+        We start with the scenario :math:`\sigma=+1`, i.e., :math:`d\leq \tfrac{N}{2}`, in which
+        case this decomposition has the form
+
+        .. math::
+
+            \Pi = \sum_{i=0}^{n-1} c_i \sum_{j=s_i}^{e_i} |j\rangle\langle j|,
+
+        where :math:`s_i` and :math:`e_i` are the start and end positions
+        for the :math:`i`\ th contribution to :math:`\Pi`.
+        Besides computing the integer decomposition itself, this decomposition function is mostly
+        about figuring out the right control structure to realize the correct positioning
+        according to :math:`s_i` and :math:`e_i`.
+
+        To begin, initialize empty lists of control wires and control values, corresponding
+        to :math:`s_0=0`. Then, starting at :math:`i=0`, which also marks the current wire,
+        go through the integer decomposition :math:`\{c_i\}_i` from above and perform the
+        following steps:
+
+          1. If :math:`c_i=0`, go to step 2 immediately. Else, we want to
+             add/subtract computational basis state projectors to the decomposition. The number of
+             projectors :math:`2^{n-1-i}` is stored in the number of control wires that we already
+             aggregated. In addition, :math:`s_i` (and thus :math:`e_i`) is encoded in the control
+             values, up to whether the projectors should be located in the :math:`|0\rangle`
+             or :math:`|1\rangle` subspace of the current target wire :math:`i`. If we want to
+             add projectors (:math:`c_i=+1`), we append at the beginning of the current subspace,
+             so in the :math:`|0\rangle` subspace. If we want to remove projectors (:math:`c_i=-1`),
+             we need to remove them from the end of the subspace, so from :math:`|1\rangle`.
+             Having figured out the subspace, we add the corresponding projector to the
+             decomposition. In code, we immediately apply the corresponding phase gate, see below.
```

Comment:
I wanted to try doing an explicit example with $d = 7 = 111_2 = 1 * 8 - 1 * 1$ and `c_i = {1, 0, 0, -1}` but I am completely lost here how even to start 😅

Reply:
This might be just me personally preferring examples, so consider this optional, but I think this could benefit from being just rewritten for an explicit example. I think most engineers would understand it from there, and can look at the code if you want to see the general perscription